### PR TITLE
Added utility for saving results to specified folder

### PR DESCRIPTION
**Pull request by ivholmlu**
# Description

Adding utility to specify path to save `_evaluations.json` and `benchmark.csv`
Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-packs/llama-index-packs-rag-evaluator/llama_index/packs/rag_evaluator/base.py [hunk 1]
```diff
@@ -64,6 +65,10 @@ def __init__(
         }
         self.eval_queue = deque(range(len(rag_dataset.examples)))
         self.prediction_dataset = None
+        if result_path is None:
+            self.result_path = "./"
+        else: 
+            self.result_path = result_path
```
Comment (nerdai) [inline llama-index-packs/llama-index-packs-rag-evaluator/llama_index/packs/rag_evaluator/base.py:71]: Nice, though do you think we can we do something a bit more robust?

Something like this: https://github.com/run-llama/llama_index/blob/7d3bb2b502c95e0677013ac935992d4cb1e36d9d/llama-index-core/llama_index/core/download/utils.py#L84

**Review by nerdai — COMMENTED** [review]
Thanks @ivholmlu. This will be a nice add. I left a comment for you in my review.

File: llama-index-packs/llama-index-packs-rag-evaluator/llama_index/packs/rag_evaluator/base.py (hunk 1 above)
Comment (ivholmlu) [reply llama-index-packs/llama-index-packs-rag-evaluator/llama_index/packs/rag_evaluator/base.py]: Made it more robust now using Path from pathlib. Also creating dir if needed

File: llama-index-packs/llama-index-packs-rag-evaluator/llama_index/packs/rag_evaluator/base.py [hunk 2]
```diff
@@ -44,6 +46,7 @@ def __init__(
         judge_llm: Optional[LLM] = None,
         embed_model: Optional[BaseEmbedding] = None,
         show_progress: bool = True,
+        result_path = None,
```
Comment (nerdai) [inline llama-index-packs/llama-index-packs-rag-evaluator/llama_index/packs/rag_evaluator/base.py:49]: missing type: Should be `Optional[str]`?

**Review by nerdai — COMMENTED** [review]
looks much improved -- thanks. Left a comment re: typing issue

**Review by logan-markewich — APPROVED** [review]

File: llama-index-packs/llama-index-packs-rag-evaluator/llama_index/packs/rag_evaluator/base.py (hunk 2 above)
Comment (ivholmlu) [reply llama-index-packs/llama-index-packs-rag-evaluator/llama_index/packs/rag_evaluator/base.py]: Good point, thx for fixing @logan-markewich

**Pull request merged**